spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -192,8 +194,10 @@
       FileScanTask task,
       Schema readSchema,
       Map<Integer, ?> idToConstant) {
+    NameMapping nameMapping = MappingUtil.create(readSchema);
```

Comment:
This shouldn't create a name mapping. It should use one from the table, if it exists.